src/main/java/ca/corefacility/bioinformatics/irida/ria/web/ajax/ClientsAjaxController.java

```diff
@@ -37,4 +46,52 @@ public ClientsAjaxController(UIClientService service) {
 		return ResponseEntity.ok(service.getClientList(request));
 	}
 
+	/**
+	 * Revoke all tokens for a client given its identifier
+	 *
+	 * @param id Identifier for a specific client
+	 */
+	@DeleteMapping("/revoke")
+	@PreAuthorize("hasRole('ROLE_ADMIN')")
+	public void deleteClientTokens(@RequestParam Long id) {
+		service.deleteClientTokens(id);
+	}
+
+	/**
+	 * Check to see if the client identifier that a user wants to use for a new client is not already used
+	 *
+	 * @param clientId Name to check if exists
+	 * @param locale   Users current locale
+	 * @return Http response indicating if the client id is valid.
+	 */
+	@RequestMapping("/validate")
+	@PreAuthorize("hasRole('ROLE_ADMIN')")
+	public ResponseEntity<String> validateClientName(@RequestParam String clientId, Locale locale) {
+		try {
+			service.validateClientId(clientId);
+			return ResponseEntity.status(HttpStatus.CONFLICT)
+					.body(messageSource.getMessage("server.AddClientForm.error", new Object[] { clientId }, locale));
+		} catch (ClientRegistrationException e) {
+			return ResponseEntity.ok("");
```

Comment:
I think this should be catching a different type of exception. I think a NoSuchClientException would be more appropriate as that is what us actually thrown in the `IridaClientDetailsServiceImpl.java` > `loadClientByClientId` method. This method signature will also need to be updated to include `throws NoSuchClientException`. What do you think?

Reply:
Makes sense to me, updated in d643207